ledger-core/application/benchs/benchrunner_test.go

```diff
@@ -177,26 +170,35 @@ func runBench(iterations int, name string, workerFunc func(ctx context.Context,
 
 			return err
 		})
+
+		select {
+		// prevent extra requests if group context is cancelled
+		case <-ctx.Done():
+			stopped = true
+		default:
+		}
+
 	}
 
 	err := g.Wait()
 	if err != nil {
 		return throw.W(err, "api failed")
 	}
 
-	<-counterStopped
```

Comment:
this will leak counting goroutine

Reply:
It should be moved closer to `return nil` but it should not be removed completly

Reply:
it will not, because it will be stopped by cancellation of ctx, and that is done by g.Wait()

Reply:
it could be stopped after this function is done and next one has already started. You need to make sure that goroutine is stopped